uqcsbot/scripts/bgg.py

```diff
@@ -0,0 +1,149 @@
+            parameters["categories"].add(i.attrib["value"])
+        # adds a mechanic
+        elif i.tag == "link" and i.attrib.get("type") == "boardgamemechanic":
+            parameters["mechanics"].add(i.attrib["value"])
+        # sets the user ratings
+        elif i.tag == "statistics":
+            for j in i[0]:
+                if j.tag == "average":
+                    parameters["score"] = j.attrib["value"]
+                if j.tag == "usersrated":
+                    parameters["users"] = j.attrib["value"]
+                if j.tag == "ranks":
+                    for k in j:
+                        if (k.attrib.get("name") == "boardgame"
+                                and k.attrib.get("value").isnumeric()):
+                            n = int(k.attrib.get("value"))
+                            o = "tsnrhtdd"[(n/10 % 10 != 1) * (n % 10 < 4) * n % 10::4]
+                            parameters["rank"] = f"{n:d}{o:s}"
+                        elif k.attrib.get("value").isnumeric():
+                            n = int(k.attrib.get("value"))
+                            o = "tsnrhtdd"[(n/10 % 10 != 1) * (n % 10 < 4) * n % 10::4]
+                            s = " ".join(k.attrib["friendlyname"].split(" ")[:-1])
+                            parameters["subranks"][s] = f"{n:d}{o:s}"
+        # sets the discription
+        elif i.tag == "description":
+            parameters["description"] = i.text
+        # sets the minimum playing time
+        elif i.tag == "minplaytime":
+            parameters["min_time"] = i.attrib["value"]
+        elif i.tag == "maxplaytime":
+            parameters["max_time"] = i.attrib["value"]
+
+    return parameters
+
+
+def format_board_game_parameters(parameters: dict) -> str:
+    message = (f"*{parameters.get('name', ':question:'):s}*\n"
+               f"A board game for {parameters.get('min_players', ':question:'):s} to"
+               + f" {parameters.get('max_players', ':question:'):s} players, with a"
+               + f" playing time of {parameters.get('min_time', ':question:'):s} minutes"
+               + ("" if parameters.get('min_time') == parameters.get('max_time')
+                  else f" to {parameters.get('max_time', ':question:'):s} minutes") + ".\n"
+               f"Rated {parameters.get('score', ':question:'):s}/10 by"
+               + f" {parameters.get('users', ':question:'):s} users.\n"
+               f"Ranked {parameters.get('rank', ':question:'):s} overall on _Board Game Geek_.\n"
+               + "".join(f"• Ranked {value:s} in the {key:s} category.\n"
+                         for key, value in parameters.get("subranks", {}).items()) +
+               f"Categories: {', '.join(parameters.get('categories', set())):s}\n"
+               f"Mechanics: {', '.join(parameters.get('mechanics', set())):s}\n"
+               f"https://boardgamegeek.com/boardgame/{parameters.get('identity', ':question:'):s}\n"
```

Comment:
Use of :question: could be confusing, esp. in a URL. Surely the ID will never be missing?

Reply:
Correct.